modules/core/04-channel/types/upgrade.go

```diff
@@ -0,0 +1,43 @@
+package types
+
+import (
+	errorsmod "cosmossdk.io/errors"
+
+	"github.com/cosmos/ibc-go/v7/internal/collections"
+)
+
+// ValidateBasic performs a basic validation of the upgrade fields
+func (u Upgrade) ValidateBasic() error {
+	if err := u.UpgradeFields.ValidateBasic(); err != nil {
+		return errorsmod.Wrap(err, "proposed upgrade fields are invalid")
+	}
+
+	if !u.Timeout.IsValid() {
+		return errorsmod.Wrap(ErrInvalidUpgrade, "upgrade timeout cannot be empty")
+	}
+
+	// TODO: determine if last packet sequence sent can be 0?
```

Comment:
Yes I think 0 should technically be a valid value (no packets sent yet), maybe @colin-axner / @AdityaSripal could confirm if we should let 0 be an accepted value here?

Reply:
[Packet sequences start at 1](https://github.com/cosmos/ibc-go/blob/main/modules/core/04-channel/keeper/handshake.go#L92). But iirc from yesterday this value can be determined within the state machine in `WriteChanUpgradeInit()` as opposed to expecting a user provided value

Reply:
Maybe we should remove this function? I think we should have `MsgUpgradeInit` and `MsgUpgradeTry` use the `UpgradeFields` and `UpgradeTimeout` directly rather than wrapping with `Upgrade`. I suggest this because it seems unnecessary to expose a field populated by the state machine to the relayer. This would render this `ValidateBasic` unnecessary as the msg should do basic validation on the fields and timeout and then the state machine would construct the upgrade type with the fields, timeout and retrieved packet sequence send. I guess we could keep if for the situation where the entire counterparty upgrade type is provided in the TRY and ACK, in this situation the sequence should not equal 0. 

My vote is to modify the init/try field for self upgrade to use the fields/timeout directly in the msg type and add a if sequence != 0 check to be used to verify the fully formed counterparty upgrade